@here/harp-mapview/lib/DecodedTileHelpers.ts

```diff
@@ -738,7 +738,12 @@ export function applyBaseColorToMaterial(
     }
 
     opacity = THREE.MathUtils.clamp(opacity, 0, 1);
-    material.opacity = opacity;
+    if (material instanceof CirclePointsMaterial || material instanceof SolidLineMaterial) {
```

Comment:
Instead of having to enumerate here all our custom materials, we can do this for any material than inherits from RawShaderMaterial, like SolidLineMaterial does. CirclePointsMaterial can be changed to derive from RawShaderMaterial too. The only difference is that we need to pass any defines explicitly, but the shader code for that material doesn't use any, so I guess it's only a matter of changing the base class.
That way we can move setOpacity to RawShaderMaterial.

Reply:
- CirclePointsMaterial now extends RawShaderMaterial
- setOpacity was moved to RawShaderMaterial
- those two instanceof checks are gone and replaced by one "instanceof RawShaderMaterial". 
- It is still possible to call `material = new CirclePointsMaterial()` without parameters, but `material.size` will not be accessible right after initialization. As discussed, in real use cases, this constructor is always called with parameters, and constructor without parameters is there just because three.js `Material.clone()` requires it, but `clone` doesn't need to access our custom properties.  I adjusted unit-tests accordingly.